### PR TITLE
1.19: Finalize change notes for JavaScript

### PR DESCRIPTION
**Pull request by felicity-semmle**
This PR prepares the JavaScript analysis change notes for inclusion in the help for LGTM Enterprise 1.19, and on help.semmle.com.

I've tried to put the edits and the text moves in separate commits:

* [Remove template text and reorganize information](https://github.com/Semmle/ql/commit/0b6cdda3522d50e96ad00d8172c78d58b1e10c2b) - no text edits
* [Text updates](https://github.com/Semmle/ql/commit/3f529e194ce6331b30101231ba2e45699453c137)
* [Move new non-LGTM queries](https://github.com/Semmle/ql/commit/fcce4616d8497d10234ec2a95b4fe013c1666edb) - moving these into a separate section. They'll be omitted from the LGTM embedded help and included on the help.semmle.com JavaScript analysis 1.19 page.
* [Add missing query ids](https://github.com/Semmle/ql/commit/a273668cf38919fbd2513bbe7ed5b12a15cecede)

There's one "TODO" - I wasn't sure which two the new queries now handle the dynamic method calls that are no longer flagged by Remote property injection (`js/remote-property-injection`).

@Semmle/js

### CODE REVIEW

**Comment by xiemaisi** [comment]
> I wasn't sure which two the new queries now handle the dynamic method calls that are no longer flagged by Remote property injection (js/remote-property-injection).

That would be `js/unsafe-dynamic-method-access` and `js/unvalidated-dynamic-method-call`.

File: change-notes/1.19/analysis-javascript.md [hunk 1]
```diff
@@ -4,74 +4,79 @@
...
-| Hard-coded data interpreted as code | security, external/cwe/cwe-506 | Highlights locations where hard-coded data is transformed and then executed as code or interpreted as an import path, which may indicate embedded malicious code ([CWE-506](https://cwe.mitre.org/data/definitions/506.html)). Results are not shown on LGTM by default. |
-| Host header poisoning in email generation |  security, external/cwe/cwe-640 | Highlights code that generates emails with links that can be hijacked by HTTP host header poisoning, indicating a violation of [CWE-640](https://cwe.mitre.org/data/definitions/640.html). Results shown on LGTM by default.  |
+| Enabling Node.js integration for Electron web content renderers (`js/enabling-electron-renderer-node-integration`) | security, frameworks/electron, external/cwe/cwe-094  | Highlights Electron web content renderer preferences with Node.js integration enabled, indicating a violation of [CWE-94](https://cwe.mitre.org/data/definitions/94.html). Results are hidden on LGTM by default. |
```
Comment (xiemaisi) [inline change-notes/1.19/analysis-javascript.md]: This query actually isn't even run on LGTM, so it should probably be moved to the "other new queries" section.

File: change-notes/1.19/analysis-javascript.md [hunk 2]
```diff
@@ -4,74 +4,79 @@
...
-| User-controlled data in file | security, external/cwe/cwe-912 | Highlights locations where user-controlled data is written to a file. Results are not shown on LGTM by default. |
+| Useless assignment to property (`js/useless-assignment-to-property`) | maintainability | Highlights property assignments whose value is always overwritten. Results are shown on LGTM by default. |
+
+## Other new queries
+
+| **Query**                                     | **Tags**                                             | **Purpose**                                                                                                                                                                 |
+|-----------------------------------------------|------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| File data in outbound network request (`js/file-access-to-http`) | security, external/cwe/cwe-200 | Highlights locations where file data is sent in a network request.|
```
Comment (xiemaisi) [inline change-notes/1.19/analysis-javascript.md]:
```suggestion
| File data in outbound network request (`js/file-access-to-http`) | security, external/cwe/cwe-200 | Highlights locations where file data is sent in a network request, possibly indicating a violation of [CWE-200](https://cwe.mitre.org/data/definitions/200.html). |
```

File: change-notes/1.19/analysis-javascript.md [hunk 3]
```diff
@@ -4,74 +4,79 @@
...
+| User-controlled data written to file (`js/http-to-file-access`) | security, external/cwe/cwe-912 | Highlights locations where user-controlled data is written to a file. |
```
Comment (xiemaisi) [inline change-notes/1.19/analysis-javascript.md]:
```suggestion
| User-controlled data written to file (`js/http-to-file-access`) | security, external/cwe/cwe-434, external/cwe/cwe-912 | Highlights locations where user-controlled data is written to a file, possibly indicating a violation of [CWE-912](https://cwe.mitre.org/data/definitions/912.html). |
```

File: change-notes/1.19/extractor-javascript.md [hunk 4]
```diff
@@ -27,13 +15,13 @@ extraction:
         - include: "**/*-min.js"
 ```
 
-## Changes to code extraction
-
-* The TypeScript compiler is now bundled with the distribution, and no longer needs to be installed manually.
-  Should the compiler version need to be overridden, set the `SEMMLE_TYPESCRIPT_HOME` environment variable to
+* The TypeScript compiler is now included in the LGTM Enterprise and Ql command-line tools installations, and you no longer need to install it manually.
```
Comment (xiemaisi) [inline change-notes/1.19/extractor-javascript.md]:
```suggestion
* The TypeScript compiler is now included in the LGTM Enterprise and QL command-line tools installations, and you no longer need to install it manually.
```

**Comment by felicity-semmle** [comment]
Thanks for the review and suggestions 😄

**Pull request merged**